### PR TITLE
[Azure Core, SQL Migration] Fix wrong endpoints being used for non-public clouds 

### PR DESCRIPTION
**Pull request by raymondtruong**
The SQL Migration extension makes use of several functions which interact with the ARM REST API, such as `getLocations`, `createResourceGroup`, `getSqlMigrationService`, `getAvailableSqlVMs`, and more. 

Some of the more common ones (such as `getLocations` and `createResourceGroup`) are provided via the Azure Core API, whereas the others that are more migration specific (such as `getSqlMigrationService` and `getAvailableSqlVMs`) are implemented by the SQL Migration extension API itself. 

An issue was reported (https://github.com/microsoft/azuredatastudio/issues/20275) which stated that non-public cloud users were unable to proceed in the SQL Migration wizard, due to the fact that these functions were trying to call the ARM REST API endpoint for the public cloud (`management.azure.com`) instead of the correct endpoint for the respective non-public clouds (the list of which can be found [here](https://github.com/microsoft/azuredatastudio/blob/main/extensions/azurecore/src/account-provider/providerSettings.ts)). Upon further investigation, both implementations -- both in Azure Core and in SQL Migration -- incorrectly default to the public cloud endpoint regardless of the account type.

In this PR, we modify the implementation of these functions to first examine the provider settings of the provided Azure account, and then make the HTTP request to the correct endpoint based on which cloud the account is a part of. This is done by exposing a new function `getProviderMetadataForAccount` in the Azure Core API. 

Tested and verified that wizard behavior now works as intended in the SQL Migration extension using a test account in USGov.
image

-- *still need to test end to end migration* --

### CODE REVIEW

**Comment by raymondtruong** [comment]
+ @brian-harris, @aasimkhan30, @alanrenmsft, @Charles-Gagnon 

FYI. I discussed these changes with Aasim offline, please review. Thanks!

File: extensions/sql-migration/src/api/utils.ts [hunk 1]
```diff
@@ -865,3 +865,21 @@ export async function getBlobLastBackupFileNamesValues(lastFileNames: azureResou
 	}
 	return lastFileNamesValues;
 }
+
+export function getCorrectArmEndpointForAccount(account: AzureAccount): string {
```
Comment (aasimkhan30) [inline extensions/sql-migration/src/api/utils.ts:869]: Can you please use the existing hosts info from here
[https://github.com/microsoft/azuredatastudio/blob/cb3ce6f1fd9d2b1185c3a483941522c336c42649/extensions/azurecore/src/account-provider/providerSettings.ts#L6](https://github.com/microsoft/azuredatastudio/blob/cb3ce6f1fd9d2b1185c3a483941522c336c42649/extensions/azurecore/src/account-provider/providerSettings.ts#L6)

**Review by aasimkhan30 — APPROVED** [review]

File: extensions/sql-migration/src/api/utils.ts [hunk 2]
```diff
@@ -865,3 +865,21 @@ export async function getBlobLastBackupFileNamesValues(lastFileNames: azureResou
...
+	// Looks up the correct ARM endpoint for the given account, as defined in extensions\azurecore\src\account-provider\providerSettings.ts
+	switch (account.properties.providerSettings.id) {
```
Comment (alanrenmsft) [inline extensions/sql-migration/src/api/utils.ts:871]: these values are already defined in the providerSettings.ts, we should reference them instead of redefining.

Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts:871]: I did this for the `getCorrectArmEndpointForAccount` implementation in azurecore, but was having some troubles doing the same for sql-migration (I think because providerSettings.ts is not defined in the same extension), but maybe I just missed something. Let me try again

Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts:871]: Yeah, I can't seem to get this working. I'm trying the exact same implementation as in azurecore, like [here](https://github.com/microsoft/azuredatastudio/pull/20304/files#diff-347d8b77019786cc29710a78e48c961ffa3ee7f3eb01d446f95f332c065d0e5bR516), using the following import for providerSettings:

`import providerSettings from '../../../azurecore/src/account-provider/providerSettings';`

But I can't get it to build properly. Maybe because this is an import from outside of this extension. Any other ideas? @alanrenmsft @aasimkhan30

Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts:871]: I'm not super familiar with imports in typescript, but that was the only way that seemed like it could have worked. If I just import azurecore then providerSettings doesn't seem to be accessible

Comment (alanrenmsft) [reply extensions/sql-migration/src/api/utils.ts:871]: i see, they are in different extensions. if you really need to know the endpoints here, you need to add it to the API that azurecore extension exposes: https://github.com/microsoft/azuredatastudio/blob/97a4ecdfbb6216c86defdaa1de2d6430b309c3a6/extensions/azurecore/src/azurecore.d.ts#L271

Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts:871]: Ah, I see. That might be the best option since it'd be nice to have this identical functionality in both azurecore and sql-migration. Let me try that.

Comment (raymondtruong) [reply extensions/sql-migration/src/api/utils.ts]: That seems to have worked, thanks @alanrenmsft. I've added a new function `getProviderMetadataForAccount` (figured it's a little more generic than the original `getCorrectArmEndpointForAccount` I defined) to the azurecore API in my latest iteration.

**Review by alanrenmsft — APPROVED** [review]
